src/Task/Git/BranchName.php

```diff
@@ -110,10 +113,18 @@ private function runMatcher(array $config, $name, $rule, $ruleName)
      */
     public function run(ContextInterface $context)
     {
-        $name = trim($this->repository->run('symbolic-ref', ['HEAD', '--short']));
         $config = $this->getConfiguration();
         $exceptions = [];
 
+        try {
+            $name = trim($this->repository->run('symbolic-ref', ['HEAD', '--short']));
+        } catch (ProcessException $e) {
+            if ($config['allow_symbolic_references'] === false) {
+                $message = "Branch naming convention task is not allowed to run on symbolic references.";
+                return TaskResult::createFailed($this, $context, $message);
+            }
```

Comment:
If the command fails and the detached head is allowed, you'll have to return a success result. Otherwise the matchers will fail.
If the detached head is not allowed, you can return the error message. Maybe it should be a bit more clear for less experienced users?